src/USER-MISC/fix_propel_self.cpp

```diff
@@ -0,0 +1,268 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* -----------------------------------------------------------------------
+   Contributed by Stefan Paquay @ Brandeis University
+
+   Thanks to Liesbeth Janssen @ Eindhoven University for useful discussions!
+----------------------------------------------------------------------- */
+
+#include "fix_propel_self.h"
+
+#include <cstdio>
+#include <cstring>
+#include <string>
+
+#include "atom.h"
+#include "atom_vec_ellipsoid.h"
+#include "citeme.h"
+#include "comm.h"
+#include "error.h"
+#include "force.h"
+#include "group.h"
+#include "math.h"
+#include "math_const.h"
+#include "math_extra.h"
+#include "math_vector.h"
+#include "modify.h"
+#include "random_mars.h"
+#include "respa.h"
+#include "update.h"
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+using namespace MathConst;
+
+#define PRINT_DEBUG_OUTPUT 0
+
+/* ---------------------------------------------------------------------- */
+
+FixPropelSelf::FixPropelSelf( LAMMPS *lmp, int narg, char **argv )
+  : Fix(lmp, narg, argv), magnitude(0.0),
+    mode(VELOCITY), n_types_filter(0), apply_to_type(NULL)
+{
+  if (narg < 5) error->all(FLERR, "Illegal fix propel/self command");
+
+  // The fix is to support the following cases:
+  // 1. Simple atoms, in which case the force points along the velocity
+  // 2. Aspherical particles with an orientation.
+  // The first argument (mode) is used to differentiate between these.
+
+  // args: fix ID all propel/self mode magnitude
+  // Optional args are
+
+  const char *mode_str = argv[3];
+
+  if (strncmp(mode_str, "velocity", 8) == 0) {
+    mode = VELOCITY;
+
+  } else if (strncmp(mode_str, "quat", 10) == 0) {
```

Comment:
10 should now be 4 ?

Reply:
These calls should not be using strncmp() at all, but strcmp(). Calling strncmp() would be wrong as it would accept incorrect keywords with additional characters. I thought, I had already corrected that in my refactoring.

Reply:
sorry for the omission, I fixed it